packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -681,10 +683,16 @@ export class Pipeline extends PipelineBase {
   private generateNameForDefaultBucketKeyAlias(): string {
     const prefix = 'alias/codepipeline-';
     const maxAliasLength = 256;
-    const uniqueId = Names.uniqueId(this);
-    // take the last 256 - (prefix length) characters of uniqueId
-    const startIndex = Math.max(0, uniqueId.length - (maxAliasLength - prefix.length));
-    return prefix + uniqueId.substring(startIndex).toLowerCase();
+    const maxUniqueIdLength = maxAliasLength - prefix.length;
+    if (FeatureFlags.of(this).isEnabled(cxapi.CODEPIPELINE_CROSS_ACCOUNT_KEY_ALIAS_STACK_SAFE_UNIQUE_ID)) {
```

Comment:
Maybe some comments would be nice here, to explain why is this behind a feature flag, and what are the behaviors in each of the branches (feature flag on/feature flag off)?

Reply:
Adding